src/vm.rs

```diff
@@ -368,7 +368,7 @@ impl<'a, S: Storage, C: PevmChain> Database for VmDb<'a, S, C> {
 
         if let Some(mut account) = final_account {
             // Check sender nonce
-            account.nonce += nonce_addition;
+            account.nonce =account.nonce.saturating_add(nonce_addition);
```

Comment:
For consistency with other places, let's skip this logic for `nonce` here. 
It has its check and returns an error on overflow anyway, so saturating is also the wrong intention.